pkg/controllers/apis/job_info.go

```diff
@@ -63,50 +63,45 @@ func (ji *JobInfo) SetJob(job *batch.Job) {
 // AddPod adds the k8s pod object values to the Pods field
 // of JobStruct if it doesn't exist. Otherwise it throws error.
 func (ji *JobInfo) AddPod(pod *v1.Pod) error {
-	taskName, found := pod.Annotations[batch.TaskSpecKey]
```

Comment:
Seems taskName is more easier to understand.

Reply:
In my opinion, `taskName` equals to name of pod which treats as task.  `specName` is a kind to distinguish from eath other's role in job, suce as master, worker, ps, chief.  I think we can be more strict on the variable name and make code have high readability.